qemu/tests/qemu_guest_agent.py

```diff
@@ -1291,6 +1291,102 @@ def gagent_check_get_disks(self, test, params, env):
+        and all of these following commands need to be checked via another
+        family command "guest-ssh-get-authorized-keys".
+        Steps:
+        1) Check basic function whether can work well.
+        2) check the result of adding new ssh keys that
+         one of them was existed.
+        3) remove ssh keys
+        4) reset ssh keys.
+        :param test: kvm test object
+        :param params: Dictionary with the test parameters
+        :param env: Dictionary with test environment.
+        """
+
+        def ssh_key_test(operation, *keys, **kwargs):
+            guest_user = params['guest_user']
+            op_func = getattr(self.gagent, 'ssh_%s_authorized_keys' % operation)
+            op_func(guest_user, *keys, **kwargs)
+            keys_ga = self.gagent.ssh_get_authorized_keys(guest_user)
+            session.cmd(params["add_line_at_end"])
+            keys_guest = session.cmd_output(cmd_guest_keys).strip()
+            _value_compared_ga_guest(keys_ga, keys_guest, operation)
+            return keys_ga, keys_guest
+
+        def _prepared_n_restore_env(status):
+            if status != "1":
+                session.cmd(params["cmd_add_user_set_passwd"])
+                session.cmd(params["set_setenforce"] % int(status))
+            else:
+                session.cmd(params["cmd_del_key_file"])
+                session.cmd(params["cmd_remove_user"])
+
+        def _generate_host_keys():
+            process.system(params["cmd_clean_keys"], shell=True)
+            status = process.system(params["ssh_keygen_cmd"],
+                                    shell=True, timeout=3)
+            if status:
+                test.error("Can not generate ssh key with no "
+                           "interaction, please have a check.")
+            cmd_get_hostkey = params["cmd_get_hostkey"]
+            host_key = process.getoutput(cmd_get_hostkey,
+                                         shell=True, timeout=3)
+            return host_key
+
+        def _value_compared_ga_guest(return_value_ga,
+                                     return_value_guest, status):
+            keys_ga_list = return_value_ga["keys"]
+            keys_guest = return_value_guest.replace("\n", ",")
+            for keys_ga in keys_ga_list:
+                if keys_ga not in keys_guest:
+                    test.fail("Key %s is not same with guest, "
+                              "%s ssh keys failed." % keys_ga, status)
+
+        session = self._get_session(params, None)
+        self._open_session_list.append(session)
+        _prepared_n_restore_env("0")
+        cmd_guest_keys = params["cmd_get_guestkey"]
+
+        error_context.context("Check the basic function ",
+                              logging.info)
+        host_key1 = _generate_host_keys()
+        ssh_key_test("add", host_key1, reset=False)
+        self.vm.wait_for_login(username=params["guest_user"],
+                               password=params["guest_user_passwd"])
```

Comment:
I guess this is not what you want, you should want to login it without the password, so should make password = "", isn't it?

Reply:
Are there any special rules for this method use? I've tried several times with this function,but still gave me this error: "Got password prompt twice    (output: "\nPermission denied, please try again.\nfedora@10.73.73.168's password: ")". host should login guest without password definitely. but it just goes against the wish. still give it more try. just update resent result here. if you can give me some advice that's more than enough for me.thanks

Reply:
@6-dehan Sorry, I just checked the method `remote.remote_login()`, which is called by `vm.wait_for_login()`, there it adds the option `-o PreferredAuthentications=password` when send `ssh` command, so even you generate keys it still asks for the password. Please go back to your original method, sorry for the inconvenience.